src/field_specials.c

```diff
@@ -3130,7 +3130,7 @@ void GetBattleFrontierTutorMoveIndex(void)
         i = 0;
         do
         {
-            if (gTutorMoves[i] == sBattleFrontier_TutorMoves2[moveIndex])
+            if (i == sBattleFrontier_TutorMoves2[moveIndex])
```

Comment:
`GetBattleFrontierTutorMoveIndex` needs to be properly adjusted for this change.

Reply:
Fixed in [54ee960](https://github.com/rh-hideout/pokeemerald-expansion/pull/2233/commits/54ee960970c31a49ebd83dfb2d396c67b2e6abc1) and found a few other oversights.